### ISSUE TITLE
```SPC-/``` executes Tcl parser tools

### ISSUE THREAD

**Issue by herbertjones** [syl20bnr/spacemacs#1158]
Tcl installs parser tools to ```/usr/bin/pt```.  Since spacemacs now checks for the executable ```pt``` to exist and prefers it to the other search options, ```SPC-/``` is broken for any Linux distribution that installs Tcl, or any user that must install it due to other packages that require it, such as the "expect" command.

**Comment by trishume** [comment]
This is a significant issue, I have the TCL pt executable as well but I worked around this problem accidentally by installing pt out of enthusiasm the moment support for it was added to spacemacs.

I guess `pt` support should be opt-in then instead of auto-detect. Everything else like `ag` doesn't have high profile collisions so those can be auto-detect.

**Comment by syl20bnr** [comment]
I like @trishume idea but I would generalize it with a variable `dotspacemacs-default-search-tool` initialized with `ag`. This variable has the effect to change what first external tool the DWIM function searches for (is this good english ? :-)).
Thus a user has to explicitly change it to `pt` so its the responsability to ensure that pt is operational in his/her environment.

**Comment by herbertjones** [comment]
Perhaps the auto-detection could be expanded to execute ```pt``` with ```--version```, and if the output doesn't match the expected output, ```pt``` is skipped as an option?  That would require more work though to get right.

**Comment by syl20bnr** [comment]
I pushed 71b68500d0b86cc4b3b48796991d60caf5760179 in `develop` and cherry-picked it in `release-0.101`.
This adds the variable `dotspacemacs-search-tools` with default value `("ag" "pt" "ack" "grep")` so "ag" will be picked first by default.
We can improve the search function by adding a check with `--version` but for now `pt --version` returns just a version number, I opened an issue to ask for more info in the output (https://github.com/monochromegane/the_platinum_searcher/issues/84).
